app/models/title.rb

```diff
@@ -7,9 +7,22 @@ class Title < RmApiResource
 
   before_request do |name, request|
     if name == :all
+      filters = request.get_params.delete(:filter) || {}
+
+      unless filters.is_a?(ActionController::Parameters) || filters.is_a?(Hash)
+        raise ActionController::BadRequest, 'Invalid filter parameter'
+      end
+
+      if filters[:selected] == 'true'
+        request.get_params[:selection] = 'selected'
+      elsif filters[:selected] == 'false'
+        request.get_params[:selection] = 'notselected'
+      elsif filters[:selected] == 'ebsco'
+        request.get_params[:selection] = 'orderedthroughebsco'
+      end
```

Comment:
Looks good - I'm adjusting my code based on some changes here as well. Wondering if we should add a default case for selection = 'all' if it is not specified?

Reply:
RM API defaults to all, but you're right, it's probably better to be explicit here.